src/releaseBusV2/release-bus-v2.reconciler.ts

```
@@ -93,8 +97,22 @@ type PreparedRepository = {
   readonly trainId: string;
-  readonly frontendRunId: string | null;
-  readonly backendRunId: string | null;
+  readonly frontend: PreparedArtifactSource | null;
+  readonly backend: PreparedArtifactSource | null;
+};
+
+type ArtifactDeployBinding = {
+  readonly artifact_environment: '' | 'staging' | 'production';
+  readonly artifact_contract_version:
+    | 'legacy-v2'
+    | typeof ENVIRONMENT_BOUND_ARTIFACT_CONTRACT;
```

ESLint

[error] Replace `|·'legacy-v2'⏎···` with `'legacy-v2'`

(prettier/prettier)